bin/github-linguist

```diff
@@ -56,10 +57,21 @@ elsif File.file?(path)
     'Binary'
   end
 
-  puts "#{blob.name}: #{blob.loc} lines (#{blob.sloc} sloc)"
-  puts "  type:      #{type}"
-  puts "  mime type: #{blob.mime_type}"
-  puts "  language:  #{blob.language}"
+  if !json_breakdown
+    puts "#{blob.name}: #{blob.loc} lines (#{blob.sloc} sloc)"
+    puts "  type:      #{type}"
+    puts "  mime type: #{blob.mime_type}"
+    puts "  language:  #{blob.language}"
+  else
+    puts JSON.generate( { blob.name =>	{
+	    					:lines => blob.loc,
+						:sloc => blob.sloc,
+						:type => type,
+						:mime_type => blob.mime_type,
+						:language => blob.language
+    					}
+    			} )
+  end
```

Comment:
Negation in the first part of an `if` statement can lead to confusion. I know we have this smelly behaviour elsewhere in this file and we can address that at a later date. For the moment, please flip this around, ie:

```ruby
if json_breakdown
  puts JSON.generate( { blob.name =>	{
  [...]
else
  puts "#{blob.name}: #{blob.loc} lines (#{blob.sloc} sloc)"
  [...]
end

Reply:
I'll refrain from making a joke about perl, here. Done.